src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs

```diff
@@ -125,7 +125,7 @@ internal XmlResolver GetXmlResolver()
         //notice we must keep GetXmlResolver() to avoid dead lock when init System.Config.ConfigurationManager
         internal XmlResolver GetXmlResolver_CheckConfig()
         {
-            if (!IsXmlResolverSet)
+            if (!LocalAppContextSwitches.DontProhibitDefaultResolver && !IsXmlResolverSet)
```

Comment:
What do you think of reversing the sense, so it's just 'ProhibitDefaultResolver'

Negative names consistently cause confusion, and worse when they are getting used in a negated context like this! 

---
In reply to: [94083128](https://github.com/dotnet/corefx/pull/14751#discussion_r94083128) [](ancestors = 94083128)